buildSrc/src/main/java/net/neoforged/neodev/installer/CreateLauncherProfile.java

```diff
@@ -0,0 +1,132 @@
+import net.neoforged.neodev.utils.DependencyUtils;
+import net.neoforged.neodev.utils.FileUtils;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.provider.ListProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.Nested;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Creates the JSON file for running NeoForge via the Vanilla launcher.
+ */
+public abstract class CreateLauncherProfile extends DefaultTask {
+    @Inject
+    public CreateLauncherProfile() {}
+
+    @Input
+    public abstract Property<String> getFmlVersion();
+
+    @Input
+    public abstract Property<String> getMinecraftVersion();
+
+    @Input
+    public abstract Property<String> getNeoForgeVersion();
+
+    @Input
+    public abstract Property<String> getRawNeoFormVersion();
+
+    @Nested
+    protected abstract ListProperty<IdentifiedFile> getLibraryFiles();
+
+    public void setLibraries(Configuration libraries) {
+        getLibraryFiles().set(IdentifiedFile.listFromConfiguration(getProject(), libraries));
+    }
+
+    @Input
+    public abstract ListProperty<URI> getRepositoryURLs();
+
+    @Input
+    public abstract ListProperty<String> getIgnoreList();
+
+    @Input
+    protected abstract Property<String> getModulePath();
+
+    public void setModules(Configuration modules) {
+        getModulePath().set(DependencyUtils.configurationToClasspath(modules, "${library_directory}/", "${classpath_separator}"));
+    }
+
+    @OutputFile
+    public abstract RegularFileProperty getLauncherProfile();
+
+    @TaskAction
+    public void createLauncherProfile() throws IOException {
+        var time = LocalDateTime.now().format(DateTimeFormatter.ISO_DATE_TIME);
+
+        getLogger().info("Collecting libraries for Launcher Profile");
+        var libraries = LibraryCollector.resolveLibraries(getRepositoryURLs().get(), getLibraryFiles().get());
+
+        var gameArguments = new ArrayList<>(List.of(
+                "--fml.neoForgeVersion", getNeoForgeVersion().get(),
+                "--fml.fmlVersion", getFmlVersion().get(),
+                "--fml.mcVersion", getMinecraftVersion().get(),
+                "--fml.neoFormVersion", getRawNeoFormVersion().get(),
+                "--launchTarget", "forgeclient"));
+
+        var jvmArguments = new ArrayList<>(List.of(
+                "-Djava.net.preferIPv6Addresses=system",
+                "-DignoreList=" + String.join(",", getIgnoreList().get()),
+                // TODO: is this still relevant in any way?
```

Comment:
I'm pretty sure it is not, the JNA libs aren't even at 5.10.0 and haven't been for a while (1.21.1 is at `5.14.0`)

Reply:
Yeah I'm also quite sure, however I was hesitant to throw this change into this PR.

Reply:
I think it's fine, there is no explainable way that it still does something

Reply:
It's not in the userdev configs anyway, so I just removed it.